### ISSUE TITLE
VS2019 16.11.2 P1.0 Compiling and Deployment problems 

### ISSUE THREAD

**Issue by gmck** [xamarin/xamarin-android#6268]
In addition to issue #6261.
 
I'm still not having any luck with deploying with 16.11.2 Prev1.0. Tt is actually getting worse, I now can't even build with it.

I went back to an older machine from my backup and it builds and deploys fine with 16.11.2 Prev1.0. I then copied the project less the obj and bin folder contents to my development machine and made two copies in different folders one for 16.11.2 Prev1.0 and one for a newly installed 16.11.2. (I've never previously had anything other than Preview versions on this computer) The 16.11.2 builds and deploys fine. I'm attaching a binlog of that build. As far as I can see that log stops after the build, you don't see any data about the deployment. And I also noticed  AndroidInstallAfterBuild = False as the last entry in the Project Graph. Since the original problem was I couldn't deploy could you please advise how you set that true.

The 16.11.2 Prev 1.0 version will not build at all. The total output is the following and the build logging never finishes - Status is Running

```
Android SDK Build Tools: C:\Program Files (x86)\Android\android-sdk\build-tools\30.0.2\
Java SDK: C:\Program Files\Android\Jdk\microsoft_dist_openjdk_1.8.0.25\
Application Java class: android.app.Application
_CleanIntermediateIfNeeded:
Creating "obj\Release\110\stamp\_CleanIntermediateIfNeeded.stamp" because "AlwaysCreate" was specified.
_ComputeAndroidResourcePaths:
Creating directory "obj\Release\110\res\".
_ResolveLibraryProjectImports:
Creating "obj\Release\110\stamp\_ResolveLibraryProjectImports.stamp" because "AlwaysCreate" was specified.
_BuildLibraryImportsCache:
Creating "obj\Release\110\stamp\_BuildLibraryImportsCache.stamp" because "AlwaysCreate" was specified.
_GenerateAndroidResourceDir:
Creating "obj\Release\110\res.flag" because "AlwaysCreate" was specified.
_GenerateLayoutBindings:
Skipping target "_GenerateLayoutBindings" because it has no outputs.
_ConvertResourcesCases:
Creating directory "obj\Release\110\flata\".
Creating directory "obj\Release\110\flat\".
Creating "obj\Release\110\stamp\_ConvertResourcesCases.stamp" because "AlwaysCreate" was specified.
_UpdateAndroidResgen:
Creating "obj\Release\110\R.cs.flag" because "AlwaysCreate" was specified.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```
I've repeated this process a couple of times over the weekend, but eventually, after making codes changes etc a build will fail to deploy again, even with the 16.11.2 version then it starts all over again and I can't compile.

As you can imagine, it is a real pain. I'd really appreciate some help with this.

[_Release_AnyCPU_Build_2021-09-05T13_43_37.6380092+10_00.zip](https://github.com/xamarin/xamarin-android/files/7110866/_Release_AnyCPU_Build_2021-09-05T13_43_37.6380092%2B10_00.zip)

**Comment by jonathanpeppers** [comment]
Visual Studio 2019 16.11.2 is the current stable release containing Xamarin.Android 11.4.0.5.

You're having some issues with `16.11.2 Prev1.0`? What version is this? Is this an old preview of Visual Studio?

**Comment by gmck** [comment]
@jonathanpeppers 
According to the help about 16.11.2 Prev 1.0 is meant to be the latest PREVIEW version. You seem to be indicating that this is an old version. As I said previously I've never had the stable release (16.11.2) on this machine until a couple of days ago.

When I compare all the Xamarin components of each version they are identical. Could you please indicate what is the latest preview version number ?

All the Xamarin stuff Xamarin, Xamarin Designer, Templates etc are the same and they both have Xamarin.Android 11.4.0.5.

When I check for updates both versions report they are up to date and neither has any notifications re updates.

I guess I should uninstall the so called preview version and reinstall it.

Normally I would expect a preview version would something like 16.11.3 plus Prev number. 

On my older machine - the release version is 16.6.2, so that gives you an idea of how far I am behind with release versions because I never really use them since I don't have a production build.

I just checked the preview version on the older machine and it too says 16.11.2 Preview 1.0 and is up to date. Its Visual Studio Installer states 16.11.2 Preview 1.0 for Preview.

Is it possible that I'm meant to be using VS 2022 to get the latest versions?

**Comment by gmck** [comment]
@jonathanpeppers, @dellis1972  

Looks like 16.11.2 is the last release of VS Studio 2019 as per these release notes https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes.

Therefore I downloaded VS 2022 Version 17.0.0 Preview 3.1 which has Xamarin.Android 11.4.99.70. That meant VS 2019 Preview had to go because I assume they can't be installed side by side.

I can now build and deploy ok. Building after a solution clean often seems to make VS 2022 hang, (Not responding message) but it does eventually recover.

If you want I can include a binlog for you to check. I need to work with it a bit and catch up on all the stuffing around I went through this last week. I'll let you know in a day or two.

Is this web page https://docs.microsoft.com/en-au/xamarin/android/release-notes/11/11.3 now obsolete?  It hasn't been updated since April. 

Is there another page we should be looking at now for the latest Xamarin.Android release notes?
